meta-lmp-base/recipes-kernel/xenomai4/xenomai4_git.bb

```diff
@@ -0,0 +1,50 @@
+SUMMARY = "Xenomai 4"
+DESCRIPTION = "Xenomai 4 user support (libevl)"
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=2500c28da9dacbee31a3f4d4f69b74d0"
+
+SRCBRANCH = "master"
+
+SRC_URI = "git://source.denx.de/Xenomai/xenomai4/libevl.git;protocol=https;branch=${SRCBRANCH}"
+SRCREV = "dbb1a86f011d7b87f1420fbf13753ea9fccef6af"
+
+S = "${WORKDIR}/git"
+
+TOOLCHAIN = "gcc"
+
+# We need the kernel sources to be available, so build it first.
+DEPENDS += "virtual/kernel"
+
+# We may need this for 64bit atomic ops.
+RDEPENDS:${PN} = "libgcc"
+
+inherit pkgconfig meson
```

Comment:
We can use the REQUIRED_MACHINE_FEATURES to be build this recipe
```
+inherit pkgconfig meson features_check
+
+REQUIRED_MACHINE_FEATURES = "xeno"
```
and when the machine feature are not enabled bitbake will fail:
```
ERROR: Nothing PROVIDES 'xenomai4'
xenomai4 was skipped: missing required machine feature 'xeno' (not in MACHINE_FEATURES)
```